vault/vault_test.go

```diff
@@ -0,0 +1,104 @@
+package vault
+
+import (
+	"github.com/hashicorp/vault/api"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+
+	"bytes"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func Test_DecryptEnvs(t *testing.T) {
+	envVault := EnvVaultImpl{client: &mockVaultAPI{}}
+
+	Convey("EnvVault", t, func() {
+		Convey("isVaultPath() returns false for a normal env value", func() {
+			value := isVaultPath("value")
+			So(value, ShouldBeFalse)
+
+			value = isVaultPath("1234")
+			So(value, ShouldBeFalse)
+
+			value = isVaultPath("http://url")
+			So(value, ShouldBeFalse)
+
+		})
+
+		Convey("isVaultPath() returns true for a vault key value", func() {
+			value := isVaultPath("vault://test")
+			So(value, ShouldBeTrue)
+
+			value = isVaultPath("vault://secret/key1")
+			So(value, ShouldBeTrue)
+		})
+
+		Convey("ReadSecretValue fails for an invalid key", func() {
+			_, err := envVault.ReadSecretValue("invalid path")
+			So(err.Error(), ShouldEqual, "Invalid Vault path 'invalid path', expecting prefix 'vault://'")
+
+		})
+
+		Convey("ReadSecretValue fails for key not found", func() {
+			_, err := envVault.ReadSecretValue("vault://secure/notFoundKey")
+			So(err.Error(), ShouldEqual, "Path 'secure/notFoundKey' not found [VAULT_ADDR:http://test]")
```

Comment:
It's not required, but I find that the tests are less brittle if you just compare for the part of the string you really need to match on vs the whole string. That way minor changes to the messages don't break the tests.